### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by LMathew2001**
### README for LSTM Configuration and Prometheus Data Fetcher
**Overview**
This project focuses on using Prometheus metrics for training an LSTM model. The provided Python script fetches data from Prometheus, processes it, and prepares it for machine learning tasks.

The project uses configuration files (lstm.conf) to manage various parameters, making it easy to customize and reuse the setup.

Files Included
Configuration File (lstm.conf)
Defines model parameters, training settings, default configurations, Prometheus queries, and filtering options.

Python Script (main.py)

Initializes Prometheus client.
Fetches data within a specific time range.
Converts Unix timestamps to a readable format.
Saves processed data to a CSV file.
Configuration (lstm.conf)
The configuration file contains the following sections:

[MODEL]
Parameters for the LSTM model:

sequence_length: Input sequence length.
lstm_units: Number of LSTM units.
gru_units: Number of GRU units.
dense_units: Units in dense layers.
dropout_rate: Dropout rate for regularization.
learning_rate: Learning rate for optimization.
hours_to_predict: Hours for prediction.
days_of_data_prom: Days of Prometheus data to use.
noise_level: Level of noise for data augmentation.
[TRAINING]
Training settings:

num_epochs: Number of training epochs.
batch_size: Training batch size.
time_duration: Time duration for training data (e.g., 1d).
step_interval: Time interval between data points (e.g., 15s).
model_expiration: Duration for model expiration.
[DEFAULT]
Default Prometheus and data paths:

prometheus_url: Prometheus API endpoint.
metric: Default metric to query.
instance_type, instance_name, namespace: Metadata for filtering Prometheus metrics.
predictions_csv_path, anomaly_file_path, checkpoint_file_path, model_file_path: Paths for output files and models.
prometheus_limit: Maximum data points fetched in a query.
[queries]
Prometheus query templates:

container_cpu_usage_seconds_total: CPU usage query.
container_memory_working_set_bytes: Memory usage query.
[FILTERING]
Filtering parameters:

iqr_multiplier: Multiplier for IQR filtering.
lower_quantile, upper_quantile: Quantiles for data filtering.
Steps to Run
Setup:

Ensure Python dependencies are installed (e.g., prometheus-api-client, pandas, tenacity, configparser, pytz).
Configure:

Modify lstm.conf to reflect your Prometheus setup and desired parameters.
Run Script:

Execute main.py to fetch and process Prometheus data:
bash
Copy code
python main.py
Output:

Processed data is saved as prometheus_data.csv.
Key Functions
initialize_prometheus_client
Connects to Prometheus and loads query templates.

fetch_prometheus_data
Retrieves metrics from Prometheus for a specific time range.

read_config
Reads and parses configurations from lstm.conf.

parse_duration_to_days
Converts time durations like 1d or 3h into days.

calculate_time_range
Computes the start and end time for data fetching.

convert_to_readable_timestamp
Converts Unix timestamps into human-readable datetime format.

Example Usage
Update the Prometheus URL, metric, and instance details in lstm.conf.
Run the script:
bash
Copy code
python main.py
View the output data in prometheus_data.csv.
Dependencies
Python 3.7+
Required Libraries:
pandas
tenacity
prometheus-api-client
configparser
pytz
Notes
Ensure Prometheus URL and queries are correctly configured.
Adjust step_interval and prometheus_limit to match your Prometheus API capabilities.
Contact
For questions or support, feel free to reach out.

### CODE REVIEW

**Comment by LMathew2001** [comment]
This project focuses on using Prometheus metrics for training an LSTM model.
